app/uk/gov/hmrc/customs/notification/controllers/HeaderValidator.scala

```diff
@@ -56,6 +55,8 @@ trait HeaderValidator {
         Future.successful(ErrorConversationIdInvalid.XmlResult)
       } else if (!hasAuth(maybeBasicAuthToken)) {
         Future.successful(ErrorUnauthorized.XmlResult)
+      } else if(!correlationIdIsValidIfPresent) {
+        Future.successful(errorBadRequest("Bad request").XmlResult)
```

Comment:
Could you just use `ErrorGenericBadRequest` here rather than `errorBadRequest("Bad request")`?